bridge/github/export.go

```diff
@@ -0,0 +1,654 @@
+				return fmt.Errorf("updating labels %v: %v", bugGithubID, err)
+			}
+
+			hash, err = opr.Hash()
+			if err != nil {
+				return fmt.Errorf("comment hash: %v", err)
+			}
+
+			id = bugGithubID
+			url = bugGithubURL
+
+		default:
+			panic("unhandled operation type case")
+		}
+
+		// mark operation as exported
+		if err := markOperationAsExported(b, hash, id, url); err != nil {
+			return fmt.Errorf("marking operation as exported: %v", err)
+		}
+	}
+
+	if err := b.CommitAsNeeded(); err != nil {
+		return fmt.Errorf("bug commit: %v", err)
+	}
+
+	return nil
+}
+
+// getRepositoryNodeID request github api v3 to get repository node id
+func getRepositoryNodeID(owner, project, token string) (string, error) {
+	url := fmt.Sprintf("%s/repos/%s/%s", githubV3Url, owner, project)
+
+	client := &http.Client{
+		Timeout: defaultTimeout,
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	// need the token for private repositories
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", token))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("error retrieving repository node id %v", resp.StatusCode)
+	}
+
+	aux := struct {
+		NodeID string `json:"node_id"`
+	}{}
+
+	data, _ := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	err = json.Unmarshal(data, &aux)
+	if err != nil {
+		return "", err
+	}
+
+	return aux.NodeID, nil
+}
+
+func markOperationAsExported(b *cache.BugCache, target git.Hash, githubID, githubURL string) error {
+	_, err := b.SetMetadata(
+		target,
+		map[string]string{
+			keyGithubId:  githubID,
+			keyGithubUrl: githubURL,
+		},
+	)
+
+	return err
+}
+
+// get label from github
+func (ge *githubExporter) getGithubLabelID(label string) (string, error) {
+	url := fmt.Sprintf("%s/repos/%s/%s/labels/%s", githubV3Url, ge.conf[keyOwner], ge.conf[keyProject], label)
+
+	client := &http.Client{
+		Timeout: defaultTimeout,
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	// need the token for private repositories
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", ge.conf[keyToken]))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != http.StatusFound {
+		return "", fmt.Errorf("error getting label: status code: %v", resp.StatusCode)
+	}
+
+	aux := struct {
+		ID     string `json:"id"`
+		NodeID string `json:"node_id"`
+		Color  string `json:"color"`
+	}{}
+
+	data, _ := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	err = json.Unmarshal(data, &aux)
+	if err != nil {
+		return "", err
+	}
+
+	return aux.NodeID, nil
+}
+
+// create github label using api v3
+func (ge *githubExporter) createGithubLabel(label, labelColor string) (string, error) {
+	url := fmt.Sprintf("%s/repos/%s/%s/labels", githubV3Url, ge.conf[keyOwner], ge.conf[keyProject])
```

Comment:
out of curiosity, why do you use the V3 API ?

Reply:
I didn't find any mutation to create label with the api v4 :X

Reply:
https://developer.github.com/v4/mutation/createlabel/ ;)